src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -117,13 +122,49 @@ namespace RSS
                     && (lastMatch == other.lastMatch)
                     && (savePath == other.savePath)
                     && (category == other.category)
-                    && (addPaused == other.addPaused);
+                    && (addPaused == other.addPaused)
+                    && (smartFilter == other.smartFilter);
         }
     };
 }
 
 using namespace RSS;
 
+QString computeEpisodeName(const QString &article)
+{
+    static const QRegularExpression episodeRegex(
+                    "(?:^|[^A-Za-z0-9])(?:"
+
+                    //Format 1: s01e01
+                    "(?:s(\\d{1,})e(\\d{1,}))|"
+
+                    //Format 2: 01x01
+                    "(?:(\\d{1,})x(\\d{1,}))|"
+
+                    //Format 3: 2017.01.01
+                    "(\\d{4}\\.\\d{2}\\.\\d{2})"
+
+                    ")(?:[^A-Za-z0-9]|$)",
```

Comment:
This is just preference, i'm just saying it in case you didn't know.
You can use `\\W` which is equivalent to `[^A-Za-z0-9_]` (IF you don't want the underscore ( _ )).
Also you can use `+` in place of `{1,}`

I don't know if format 3 is a standard but would there be any chance that it might say e.g `2015.3.15` (omit leading zeros)? Also could it use slashes or minus/dash instead of dots?